src/utils/k_fold_cv.py

```diff
@@ -0,0 +1,99 @@
+import hydra
+from copy import deepcopy
+from typing import List
+import string
+import random
+import uuid
+
+import pytorch_lightning
+from pytorch_lightning import Trainer, LightningModule, Callback
+from pytorch_lightning.callbacks import ModelCheckpoint
+from pytorch_lightning.loggers import LoggerCollection, LightningLoggerBase
+
+import wandb
+import numpy as np
+
+from src.utils import utils
+
+log = utils.get_logger(__name__)
+
+
+class CV:
+    """Cross-validation with a LightningModule."""
+    def __init__(self,
+                 config):
+        super().__init__()
+        self.config = config
+
+    @staticmethod
+    def update_modelcheckpoint(model_ckpt_callback, fold_idx):
+        _default_filename = '{epoch}-{step}'
+        _suffix = f'_fold{fold_idx}'
+        if model_ckpt_callback.filename is None:
+            new_filename = _default_filename + _suffix
+        else:
+            new_filename = model_ckpt_callback.filename + _suffix
+        setattr(model_ckpt_callback, 'filename', new_filename)
+
+    @staticmethod
+    def init_logger(config, name: str, fold_idx: int):
+        logger: List[LightningLoggerBase] = []
+        if "logger" in config:
+            for _, lg_conf in config.logger.items():
+                if "_target_" in lg_conf:
+                    log.info(f"Instantiating logger <{lg_conf._target_}>")
+                    logger.append(hydra.utils.instantiate(lg_conf, name=name + f'_fold_{fold_idx + 1}'))
+
+        return logger
+
+    @staticmethod
+    def init_callback(config):
+        callbacks: List[Callback] = []
+        if "callbacks" in config:
+            for _, cb_conf in config.callbacks.items():
+                if "_target_" in cb_conf:
+                    log.info(f"Instantiating callback <{cb_conf._target_}>")
+                    callbacks.append(hydra.utils.instantiate(cb_conf))
+        return callbacks
```

Comment:
If the code fragment is repeated in `train.py`  then it should probably be abstracted away.